numpy/core/include/numpy/ndarraytypes.h

```diff
@@ -1225,6 +1225,8 @@ struct PyArrayIterObject_tag {
                 _PyAIT(it)->dataptr = PyArray_BYTES(_PyAIT(it)->ao); \
                 for (__npy_i = 0; __npy_i<=_PyAIT(it)->nd_m1; \
                      __npy_i++) { \
+                        _PyAIT(it)->coordinates[__npy_i] = \
+                                (__npy_ind / _PyAIT(it)->factors[__npy_i]); \
```

Comment:
Sorry, shouldn't hvae "dismissed the review".  This had looked a bit weird to me with all the division (and I missed the modulo later).  I still wonder if there isn't a nicer way to do this. But thats a different issue :).

Reply:
I copied what GOTO does.